oauthenticator/github.py

```diff
@@ -27,6 +28,30 @@
 else:
     GITHUB_API = '%s/api/v3/user' % GITHUB_HOST
 
+
+def _api_headers(access_token):
+    return {"Accept": "application/json",
+            "User-Agent": "JupyterHub",
+            "Authorization": "token {}".format(access_token)
+           }
+
+
+def _get_next_page(response):
+    # Github uses Link headers for pagination.
+    # See https://developer.github.com/v3/#pagination
+    link_header = response.headers.get('Link')
+    if not link_header:
+        return
+    # parse the Link header; is this regex general enough?
+    link_regex = r',? *< *(.*?) *> *; *rel="(.*?)"'
```

Comment:
To avoid maintaining our own implementation, we could vendor requests' [own implementation of this](https://github.com/kennethreitz/requests/blob/v2.13.0/requests/utils.py#L670):

```python
for link in parse_header_links(link_header):
    if link.get('rel') == 'next':
        return link['url']
```